src/lib/react/ReactMacro.hx

```diff
@@ -295,6 +295,37 @@ class ReactMacro
 		return fields;
 	}
 
+	static function buildJsxStatic():Array<Field> {
+		#if debug
+		var localClassRef:Null<Ref<ClassType>> = Context.getLocalClass();
+		if (localClassRef == null) return null;
+		var localClass = localClassRef.get();
+
+		var jsxStaticField = extractMetaString(localClass.meta, ":jsxStatic");
+		if (jsxStaticField == null) return null;
+
+		var pos = Context.currentPos();
+		var className = localClass.name;
+		var classPath = localClass.pack.concat([className]).join("_");
+
+		var field:Field = {
+			name: '__displayName',
+			access: [Access.AStatic, Access.APrivate],
+			kind: FieldType.FVar(null, macro {
+				untyped $i{classPath}.$jsxStaticField
```

Comment:
I'm afraid this won't be compatible with code splitting - what does the generated JS look like?

To work with Modular it should be added in static `_init_` and the code should be a direct assignement:
```
com_Something.StaticField.displayName = 'StaticField';
```
Though we'd need to check but maybe this will work as well (in `_init_`):
```
if (com_Something.StaticField != null)
    com_Something.StaticField.displayName = 'StaticField';
```

Reply:
This adds assignements alongside `tagComponent()` output, at the end of the generated js:

```javascript
components_Footer.__displayName = components_Footer.render ? components_Footer.render.displayName = "Footer" : false;
components_Link.displayName = "Link"; // From tagComponent()
components_Todo.__displayName = components_Todo.render ? components_Todo.render.displayName = "Todo" : false;
```

I'm not sure about code splitting, but it seems to work (or not to) like existing code. I'd need to check that.

Reply:
Ok that would be compatible with code splitting.